calculation/seistech_calc/seistech_calc/utils.py

```diff
@@ -418,3 +420,244 @@ def to_mu_sigma(df: pd.DataFrame, im: IM):
     return df.loc[:, [str(im), f"{im}_sigma"]].rename(
         columns={str(im): "mu", f"{im}_sigma": "sigma"}
     )
+
+
+def calculate_gms_spectra(gms_data: Dict, num_gms: int):
```

Comment:
I'm not 100% sure what this function does, however it seems quite complex. Might be worth for us to go through this together and see if we can simplify it.

Reply:
I assume its a pretty direct port from the javascript frontend code?